lib/improver/nbhood.py

```diff
@@ -456,7 +459,7 @@ def process(self, cube):
             for cube_slice, radius_in_km in (
                     zip(cube.slices_over("time"), required_radii_in_km)):
                 cube_slice = self.neighbourhood_method(
-                    cube_slice, radius_in_km, self.unweighted_mode).run()
+                    radius_in_km, self.unweighted_mode).run(cube_slice)
                 cube_slice = iris.util.new_axis(cube_slice, "time")
```

Comment:
As I think that you have a valid point regarding creating a new object for each `cube_slice`, I've made these edits as you've suggested.